Explorer/Assets/DCL/Multiplayer/Movement/Systems/SDKAvatarShapesMotionSystem.cs

```diff
@@ -0,0 +1,108 @@
+        {
+            UpdateMotionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(Profile))]
+        private void UpdateMotion(
+            in IAvatarView view,
+            ref CharacterTransform characterTransformComponent,
+            ref CharacterLastPosition characterLastPositionComponent,
+            ref CharacterAnimationComponent animationComponent)
+        {
+            UpdateRotation(characterTransformComponent, ref characterLastPositionComponent);
+            UpdateAnimations(view, characterTransformComponent, ref characterLastPositionComponent, ref animationComponent);
+            UpdateLastPosition(ref characterTransformComponent, ref characterLastPositionComponent);
+        }
+
+        private static void UpdateRotation(
+            CharacterTransform characterTransformComp,
+            ref CharacterLastPosition characterLastPositionComp)
+        {
+            var currentPosition = characterTransformComp.Transform.position;
+            var oldPosition = characterLastPositionComp.LastPosition;
+
+            if (oldPosition == currentPosition)
+                return;
+
+            Vector3 newDirection = currentPosition - oldPosition;
+            newDirection.y = 0;
+
+            if (newDirection != Vector3.zero)
+            {
+                Vector3 currentForward = characterTransformComp.Transform.forward;
+                Vector3 targetForward = newDirection.normalized;
+                characterTransformComp.Transform.forward = Vector3.Lerp(currentForward, targetForward, UnityEngine.Time.deltaTime * ROTATION_SPEED);
+            }
+        }
+
+        private static void UpdateAnimations(
+            IAvatarView view,
+            CharacterTransform characterTransformComp,
+            ref CharacterLastPosition characterLastPositionComp,
+            ref CharacterAnimationComponent animationComponent)
+        {
+            const bool IS_JUMP_TRIGGERED = false;
+            const bool IS_STUNNED = false;
+
+            var currentPosition = characterTransformComp.Transform.position;
+            var oldPosition = characterLastPositionComp.LastPosition;
+            float movementBlendValue = 0;
+
+            if (oldPosition != currentPosition)
+            {
+                float speed = Vector3.Distance(oldPosition, currentPosition) / UnityEngine.Time.deltaTime;
+                movementBlendValue = speed > RUN_SPEED_THRESHOLD ? 3 : speed > JOG_SPEED_THRESHOLD ? 2 : 1;
+            }
+
+            animationComponent.IsSliding = false;
+            animationComponent.States.MovementBlendValue = movementBlendValue;
+            animationComponent.States.SlideBlendValue = 0;
+            animationComponent.States.IsGrounded = true;
+            animationComponent.States.IsJumping = false;
+            animationComponent.States.IsLongJump = false;
+            animationComponent.States.IsLongFall = false;
+            animationComponent.States.IsFalling = false;
```

Comment:
can be probably extracted to some function like `SetGroundedMovement`

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/3297/commits/aa7a82be1a444750811e9e11429f00edfc9b59b4